agent/src/agent/pipeline/pipeline.py

```diff
@@ -30,9 +32,9 @@ def file_path(self) -> str:
     def to_dict(self):
         return {
             **self.config,
+            self.OVERRIDE_SOURCE: self.override_source,
             'pipeline_id': self.id,
-            'source': self.source.to_dict() if self.source else None,
-            'destination': self.destination.to_dict()
+            'source_name': self.source.name,
```

Comment:
Backward compatibility will be broken if you add source_name instead of ['source']['name']